components/brave_rewards/browser/file_util.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/file_util.h"
+
+#include <memory>
+
+#include "base/logging.h"
+
+namespace brave_rewards {
+
+namespace {
+
+const int64_t kChunkSize = 1024;
+
+int64_t SeekNumLines(
+    base::File* file,
+    const uint64_t num_lines) {
+  DCHECK(file);
+
+  if (num_lines == 0) {
+    return 0;
+  }
+
+  int64_t length = file->GetLength();
+  if (length == -1) {
+    return -1;
+  }
+
+  if (length == 0) {
+    return 0;
+  }
+
+  uint64_t line_count = 0;
+
+  char chunk[kChunkSize];
+  int64_t chunk_size = kChunkSize;
+  int64_t last_chunk_size = 0;
+
+  if (file->Seek(base::File::FROM_END, 0) == -1) {
+    return -1;
+  }
+
+  do {
+    if (chunk_size > length) {
+      chunk_size = length;
+    }
```

Comment:
Seems like this isn't really part of the loop as such. Maybe it would make a little more sense moved to where we initialize `chunk_size`?

Reply:
This is required to make sure the chunk size for EOF does not exceed the remaining bytes

Reply:
Yep, I see